roles/ceph-common/handlers/main.yml

```diff
@@ -1,8 +1,12 @@
 ---
+# upgrading ceph from hammer to jewel will make the restarting failed,
+# this will cause ceph cluster down.
+# this happens when operater really upgrades ceph and forget to set ceph.upgrade2jewel=true
+# so we can't set default value of ceph.upgrade2jewel as false here
 - name: restart ceph mons
   service: name=ceph-mon-all state=restarted
-  when: ceph_socket.rc == 0 and not ceph.scaleout|default('False')|bool
+  when: ceph_socket.rc == 0 and not ceph.scaleout|default('False')|bool and not ceph.upgrade2jewel|bool
```

Comment:
@j2sol @nirajdp76 , ceph upgrade need two steps:
`1. install new ceph packets
2. change owner of ceph dir, and restart ceph`
using run-time fact doesn't work in follow case:
The first step is successful, but the deploy fails before the second step. Then we rerun the deploy, ansible will find that the desired ceph version is the same with current version. It will not execute the second step of ceph upgrade.